client/messaging/external-link.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { useAppDispatch } from '../redux-hooks'
+import { maybeOpenExternalLinkDialog } from './action-creators'
+
+interface ExternalLinkProps {
+  href: string
+  innerText: string
+}
+
+export default function ExternalLink({ href, innerText }: ExternalLinkProps) {
```

Comment:
Doesn't seem like this component is necessary anymore. You could attach the `onClick` event handler directly on the anchor element in the `TextMessage`.

Only potential issue is that using `useAppDispatch` hook might break tests without doing some kind of mocking. If that's the case, then ignore this comment and just leave this component as it is.

Reply:
>If that's the case

It is: `could not find react-redux context value; please ensure the component is wrapped in a <Provider>`. So I leave component then.